addons/mrp/models/stock_move.py

```diff
@@ -291,6 +291,11 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     def write(self, vals):
+        if 'product_id' in vals and self.product_id.id != vals.get('product_id'):
```

Comment:
please add a check on the move to test it's byproduct. Then copy the move, unlink the original and action_confirm() the copy. 
We should try to not manage stock move line here

Reply:
Hello @Whenrow 
Changes has been done please check.
Thanks!

Reply:
@jipr-odoo Could you add an `if` statement on the `byproduct_id` fields ? and also one on the state being different from draft, done or cancel

Reply:
Hey @Whenrow   can you tell me in which use case the byproduct_id is used? I've checked some test cases, and I always get an empty value for byproduct_id.

Reply:
https://github.com/odoo/odoo/blob/1537161afee30714b9b04fc5c0c4e8de9348f880/addons/mrp/models/mrp_production.py#L1005-L1023
you can see the byproduct move are updated with the byproduct key

Reply:
Hello @Whenrow 
Changes are done please check.
Thanks!